File: criptografar.java
public class criptografar {
    public static void main(String[] args) {
// Definindo a mensagem e a chave de criptografia
        String mensagem = "SEGURANÇA";
        int chave = 3;

        // Criptografando a mensagem
        String mensagemCriptografada = criptografar(mensagem, chave);
        System.out.println("Mensagem criptografada: " + mensagemCriptografada);

        // Descriptografando a mensagem criptografada
        String mensagemDescriptografada = descriptografar(mensagemCriptografada, chave);
        System.out.println("Mensagem descriptografada: " + mensagemDescriptografada);

        // Definindo a palavra e outra chave de criptografia
        String palavra = "MARIAS";
        int outraChave = 4;

        // Criptografando a palavra
        String palavraCriptografada = criptografar(palavra, outraChave);
        System.out.println("Palavra criptografada: " + palavraCriptografada);

        // Descriptografando a palavra criptografada
        String palavraDescriptografada = descriptografar(palavraCriptografada, outraChave);
        System.out.println("Palavra descriptografada: " + palavraDescriptografada);
    }

    // Método para criptografar uma mensagem
    public static String criptografar(String mensagem, int chave) {
        StringBuilder criptografada = new StringBuilder();
        for (int i = 0; i < mensagem.length(); i++) {
            char letra = mensagem.charAt(i);
            if (letra >= 'A' && letra <= 'Z') { // Verifica se o caractere é uma letra maiúscula
                // Criptografa a letra movendo-a por 'chave' posições no alfabeto
                char novaLetra = (char) ((letra - 'A' + chave) % 26 + 'A');
                criptografada.append(novaLetra);
            } else {
                criptografada.append(letra); // Se não for uma letra maiúscula, mantém o caractere original
            }
        }
        return criptografada.toString(); // Retorna a mensagem criptografada
    }

    // Método para descriptografar uma mensagem criptografada
    public static String descriptografar(String mensagemCriptografada, int chave) {
        StringBuilder descriptografada = new StringBuilder();
        for (int i = 0; i < mensagemCriptografada.length(); i++) {
            char letra = mensagemCriptografada.charAt(i);
            if (letra >= 'A' && letra <= 'Z') { // Verifica se o caractere é uma letra maiúscula
                // Descriptografa a letra movendo-a por '-chave' posições no alfabeto
                char novaLetra = (char) ((letra - 'A' - chave + 26) % 26 + 'A');
                descriptografada.append(novaLetra);
            } else {
                descriptografada.append(letra); // Se não for uma letra maiúscula, mantém o caractere original
            }
        }
        return descriptografada.toString(); // Retorna a mensagem descriptografada
    }
}

File: criptografar.py
def criptografar(mensagem, chave):
    mensagem_criptografada = ""
    for letra in mensagem:
        if letra.isalpha():  # Verifica se o caractere é uma letra
            if letra.isupper():  # Verifica se a letra é maiúscula
                # Criptografa a letra maiúscula utilizando a cifra de César
                nova_letra = chr((ord(letra) - ord('A') + chave) % 26 + ord('A'))
            else:
                # Criptografa a letra minúscula utilizando a cifra de César
                nova_letra = chr((ord(letra) - ord('a') + chave) % 26 + ord('a'))
        else:
            # Mantém o caractere original se não for uma letra
            nova_letra = letra
        mensagem_criptografada += nova_letra
    return mensagem_criptografada

def descriptografar(mensagem_criptografada, chave):
    mensagem_descriptografada = ""
    for letra in mensagem_criptografada:
        if letra.isalpha():  # Verifica se o caractere é uma letra
            if letra.isupper():  # Verifica se a letra é maiúscula
                # Descriptografa a letra maiúscula utilizando a cifra de César
                nova_letra = chr((ord(letra) - ord('A') - chave + 26) % 26 + ord('A'))
            else:
                # Descriptografa a letra minúscula utilizando a cifra de César
                nova_letra = chr((ord(letra) - ord('a') - chave + 26) % 26 + ord('a'))
        else:
            # Mantém o caractere original se não for uma letra
            nova_letra = letra
        mensagem_descriptografada += nova_letra
    return mensagem_descriptografada

# Exemplo de uso
mensagem = "SEGURANÇA"
chave = 3

mensagem_criptografada = criptografar(mensagem, chave)
print("Mensagem criptografada:", mensagem_criptografada)

mensagem_descriptografada = descriptografar(mensagem_criptografada, chave)
print("Mensagem descriptografada:", mensagem_descriptografada)

palavra = "MARIAS"
outra_chave = 4

palavra_criptografada = criptografar(palavra, outra_chave)
print("Palavra criptografada:", palavra_criptografada)

palavra_descriptografada = descriptografar(palavra_criptografada, outra_chave)
print("Palavra descriptografada:", palavra_descriptografada)
